orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db_compressed.sh

```diff
@@ -0,0 +1,254 @@
+    errorExit "Unable to re-write file $2"
+  else
+    rm "${2}_original"
+  fi
+}
+
+move_backup_to_cloud () {
+  LOCAL_BACKUP="$1"
+  REMOTE_BACKUP_NAME="$2"
+  run_kinit
+  doLog "INFO Uploading to ${BACKUP_LOCATION}"
+
+  hdfs --loglevel ERROR dfs -mkdir -p "$BACKUP_LOCATION"
+  OBJECT_STORE_PATH="${BACKUP_LOCATION}/${REMOTE_BACKUP_NAME}"
+  hdfs --loglevel ERROR dfs -moveFromLocal -f "$LOCAL_BACKUP" "$OBJECT_STORE_PATH" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to upload $SERVICE backup"
+  doLog "INFO Completed upload to ${BACKUP_LOCATION}"
+}
+
+close_existing_connections() {
+  SERVICE=$1
+  doLog "INFO Closing existing connections to ${SERVICE}"
+  psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity WHERE pg_stat_activity.datname = '${SERVICE}' AND pid <> pg_backend_pid();" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to close connections to ${SERVICE}"
+}
+
+limit_incomming_connection() {
+  SERVICE=$1
+  COUNT=$2
+  doLog "INFO limit existing connections to ${COUNT} on ${SERVICE}"
+  psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "alter user ${SERVICE} connection limit ${COUNT};" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to limit connections to ${SERVICE}"
+}
+
+is_database_exists() {
+  SERVICE=$1
+  doLog "INFO Checking the existence of database ${SERVICE}"
+  database_name=$(psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "SELECT datname FROM pg_catalog.pg_database WHERE datname = '$SERVICE';" -At > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to check the existence of database $SERVICE")
+  if [[ "$database_name" != "$SERVICE" ]];then
+    return 1
+  fi
+  return 0
+}
+
+backup_database_for_service() {
+  SERVICE="$1"
+  is_database_exists $SERVICE
+  if [ "$?" -eq 1 ];then
+    doLog "WARN database for $SERVICE doesn't exist"
+    return 0
+  fi
+
+  if [[ "$CLOSECONNECTIONS" == "true" ]]; then
+    limit_incomming_connection $SERVICE 0
+    close_existing_connections $SERVICE
+  fi
+
+  doLog "INFO Dumping ${SERVICE}"
+  LOCAL_BACKUP=${DATE_DIR}/${SERVICE}_backup
+  if [[ "$SERVICE" == "ranger" ]]; then
+    REMOTE_BACKUP_NAME="ranger_backup"
+    doLog "INFO Do not compress the 'ranger' database"
+    pg_dump --host="$HOST" --port="$PORT" --username="$USERNAME" --dbname="$SERVICE" --format=plain --file="$LOCAL_BACKUP" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to dump ${SERVICE}"
+    replace_ranger_group_before_export $RANGERGROUP $LOCAL_BACKUP
+  else
+    REMOTE_BACKUP_NAME="${SERVICE}_backup.dump"
+    pg_dump --host="$HOST" --port="$PORT" --username="$USERNAME" --dbname="$SERVICE" --format=custom --file="$LOCAL_BACKUP" $COMPRESSION > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to dump ${SERVICE}"
```

Comment:
:point_up:  original script dumps in plain-sql format which is imported by `psql`. This version dumps into postgres own "custom" format which is restored by `pg_restore`. It is compressed by default and supports parallel restoration.

Reply:
xisting script does not support adding new parameters so changing it would break backward compatibility.